### DISCUSSION TITLE
FastStream Altering Existing JetStream's Stream Settings

### DISCUSSION THREAD

**Discussion by Atanusaha143** [ag2ai/faststream#1891] [Q&A] [answered]
I created a stream with specific properties using the CLI. However, when I tested a subscriber using [FastStream](https://faststream.airt.ai/latest/), it unexpectedly altered the existing stream’s settings. Why did this happen?

https://github.com/user-attachments/assets/3b692de7-8233-4745-975f-d44adce1ff37

**Comment by Lancetnik** [comment, marked as answer]
@Atanusaha143 **FastStream** tries to create / update stream by default
Please, use stream as `@broker.subscriber(..., stream=JStream("test_pull_stream", declare=False))` to prevent this behavior

**Reply by Atanusaha143** [reply]
Thanks @Lancetnik 👍🏻
